app.ts

```diff
@@ -0,0 +1,261 @@
+            class="navigation-input"
+            value="0"
+          />
+        </div>
+        <p class="amount-of-records"></p>
+      </div>
+    <button id="get-record-btn">See Record</button>
+    `;
+
+  selectionArea.innerHTML = "";
+  selectionArea.innerHTML = singleRecordSelection;
+
+  let returnBtn: any = document.querySelector("#return-btn");
+  let recordIdValue: any = document.querySelector("#record-id.navigation-input");
+  let numberOfRows = Math.floor(window.innerHeight / 50);
+
+  returnBtn.addEventListener("click", () => {
+    createNavigation();
+    let currentPage: any = document.querySelector(".current-page");
+    currentPage.innerHTML = `${fromNumber}  / ${fromNumber + numberOfRows}.`;
+  });
+
+  let getSingleRecord: any = document.querySelector("#get-record-btn");
+
+  getSingleRecord.addEventListener("click", () => {
+    fromNumber = Number(recordIdValue.value);
+    let toNumber = fromNumber + numberOfRows;
+    let check = ["undefined", "string", ""];
+
+    if (check.includes(typeof fromNumber) || fromNumber < 0) {
+      alert("Does not exists");
+      recordIdValue.value = "0";
+    } else if (typeof fromNumber === "number" && fromNumber >= 0) {
+      getRecords(fromNumber, toNumber);
+    } else {
+      alert("Error");
+    }
+  });
+}
+
+function createHeadingGrid(headings: any) {
+  let headingsData: any = `<h1 class="column-heading">${headings}</h1>`;
+  headingColumns.innerHTML += headingsData;
+}
+
+let headingRowCreation = () => {
+  fetch("http://localhost:2050/columns", {
+    method: "GET",
+    headers: { "Content-Type": "application/json" },
+  })
+    .then((response) => response.text())
+    .then((data) => {
+      let headingDataList = JSON.parse(data);
+      let headings: string;
+
+      for (let i = 0; i < headingDataList.length; i++) {
+        headings = headingDataList[i];
+        createHeadingGrid(headings);
+      }
+      resizeScreenData();
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+function dynamicGrid(columnData: any) {
+  // Creates the row that the info will display and adds it to the infoColumnsArea.
+  let infoDataRow = `<div id="info-row-${columnData[0]}" class="info-rows"></div>`;
+  infoColumns.innerHTML += infoDataRow;
+  // Gets the created rows.
+  let finalInfoDataRow: any = document.querySelector("#info-row-" + columnData[0] + ".info-rows");
+
+  // Loops through
+  for (let x = 0; x < columnData.length; x++) {
+    let infoData = `<p class="info-row-data">${columnData[x]}</p>`;
+    finalInfoDataRow.innerHTML += infoData;
+  }
+}
+
+let resizeScreenData = () => {
+  let toNumber: number;
+  let selectionArea: any = document.querySelector("#record-navigation-container");
+
+  if (selectionArea.contains(document.querySelector(".navigation-btns"))) {
+    let numberOfRows = Math.floor(window.innerHeight / 50);
+
+    nextBtn.disabled = false;
+    previousBtn.disabled = false;
+
+    if (fromNumber + numberOfRows >= 999999) {
```

Comment:
I only noticed now you are working on a hardcoded value for the max recordCount (999999). Please ask the API once for that value and store it. You can then use it accordingly instead of the working of a hardcoded values.